sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -94,6 +94,10 @@ private[sql] trait LookupCatalog extends Logging {
    * Extract catalog and identifier from a multi-part name with the current catalog if needed.
    * Catalog name takes precedence over identifier, but for a single-part name, identifier takes
    * precedence over catalog name.
+   *
+   * Note that, this pattern is used to look up permanent catalog objects like table, view,
+   * function, etc. If you need to look up temp objects like temp view, please do it separately
+   * before calling this pattern, as temp objects don't belong to any catalog.
```

Comment:
The term `Catalog` is being used in many places. For example, our internal SessionCatalog also manages temp objects.

Reply:
> our internal SessionCatalog also manages temp objects

We can move them out and put it in a temp view manager like the `GlobalTempViewManager`. I'm talking more about the theory.